tokio-quiche/src/quic/router/mod.rs

```diff
@@ -452,7 +452,18 @@ where
                         let mut gro = None;
                         let mut dst_addr_override = None;
 
-                        for cmsg in r.cmsgs() {
+                        let Ok(cmsgs) = r.cmsgs() else {
+                            // Best-effort if we can't read cmsgs.
+                            return Poll::Ready(Ok(PollRecvData {
+                                bytes,
+                                src_addr: peer_addr,
+                                dst_addr_override,
+                                rx_time,
+                                gro,
+                            }));
```

Comment:
> Best-effort

As in we attempt to read the cms and if we can't then we just skip reading? Is this the correct.. as in should we fail here instead rather than silently passing?

Can you help me understand what functionality will be missing if we return here? We might want to keep the behavior but document it better.